public/Aryan Suri/Citation/citation.js

```diff
@@ -0,0 +1,225 @@
+                if ((/^[a-z]/g).test(arname[j]) && lnameind == arname.length - 1) {
+                    lnameind = j;
+                }
+                if (j < lnameind) {
+                    fname += arname[j] + ' ';
+                } else {
+                    lname += arname[j] + ' ';
+                }
+            }
+            fname = fname.slice(0, fname.length - 1);
+            lname = lname.slice(0, lname.length - 1);
+            namedict = {
+                'given': fname,
+                'family': lname
+            }
+        }
+        authors.push(namedict)
+    }
+
+    if (verbose) {
+        console.log(rawnames);
+        console.log("Split Indices Found: " + splind);
+        console.log("Raw Author Names Found: " + rawauthnames);
+        console.log("Authors Found: ");
+        for (var i in authors) {
+            console.log(i);
+            console.log("Given: " + authors[i]['given']);
+            console.log("Family: " + authors[i]['family']);
+        }
+    }
+    console.log(authors);
+    return authors
+}
+
+
+function buildcite() {
+
+    let citeDiv = document.createElement("div");
+    $(citeDiv).attr("id", "SB-PC-AD");
+    $('body').append(citeDiv);
+
+    $(citeDiv).html(`
+
+<div onclick="hidecite()" id="citeModal">
+
+       <div id="citeModalContent" style="cursor: pointer" >
+            <div style= "min-height: 100px;" id="citeHTML">
+                <p id="citeText"> </p>
+            </div>
+
+            <div>
+                <select id="citeSelect">
+                    <option value="" selected="true" disabled="true">Template Selection</option>
+                    <option value="citation-apa">APA</option><option value="harvard1">Harvard</option>
+                    <option value="chicago">Chicago</option>
+                    <option value="vancouver">Vancouver</option>
+                    <option value="mla">MLA</option>
+                    <option value="acs">ACS</option>
+                    <option value="bibtex">BibTeX</option>
+                </select>
+            </div>
+            <div id="citeContent">
+                 <a id="citeCopy" style="text-decoration: none; color: #666" >Copy Text</a>
+                 <a id="citeCopyHTML" style="text-decoration: none; color: #666" >Copy HTML</a>
+            </div>
+       </div>
+</div>
+`);
+    $('#citeSelect').val('citation-apa')
+    getCite();
+    $('#citeSelect').on('change', function () { getCite() });
+    $('#citeBIBTEX').on('click', function () { getBibTeX() });
+    const citeCopy = document.getElementById("citeCopy");
+    citeCopy.addEventListener("click", function () {
+        let text = document.getElementById("citeText").innerText;
+        let elem = document.createElement("textarea");
+        document.body.appendChild(elem);
+        elem.value = text;
+        elem.select();
+
+        document.execCommand("copy");
+        document.body.removeChild(elem);
+    });
```

Comment:
the copy warning comes up anyways, it would be to obtrusive

Reply:
Hmm should the copy warning come up if you are simply referencing something or attributing? Once this new version is live, I would recommend asking for feedback about this since I personally feel like it is too strong for an attribution/citation.

Reply:
not if you are referencing something, only if you copy the citation/attribution to clipboard

Reply:
Clarification: By referencing I meant using the citation as a reference, which implicity requires copying the citation to use as a reference.

I am unsure if the Copy Warning should come up if copying the attribution/citation to the clipboard. When you have people test this once this is released, I would recommend you ask for their feedback specifically on this issue.